march_gait_selection/src/march_gait_selection/transition_gait/create_transition.py

```diff
@@ -0,0 +1,206 @@
+                                                   transition_velocities, transition_durations)
+
+        return self._new_subgait(self._robot, subgait_dict)
+
+    @staticmethod
+    def _scale_subgait_duration(old_subgait, new_subgait):
+        """Scale the subgaits to have matching setpoints and corresponding timestamps.
+
+        :param old_subgait:
+            The subgait which the transition is starting from
+        :param new_subgait:
+            The subgait which the transition will result in.
+
+        :return:
+            Same subgaits but with matching amount of setpoints and matching timestamps
+        """
+        timestamps = sorted(set(old_subgait.get_unique_timestamps() + new_subgait.get_unique_timestamps()))
+
+        for old_joint in old_subgait.joints:
+            new_joint = new_subgait.get_joint(old_joint.name)
+
+            old_joint_setpoints = []
+            new_joint_setpoints = []
+            for timestamp in timestamps:
+                old_joint_setpoints.append(old_joint.get_interpolated_setpoint(timestamp))
+                new_joint_setpoints.append(new_joint.get_interpolated_setpoint(timestamp))
+
+            old_joint.setpoints = old_joint_setpoints
+            new_joint.setpoints = new_joint_setpoints
+
+        return old_subgait, new_subgait
+
+    @staticmethod
+    def _transition_trajectory(old_subgait, new_subgait):
+        """Calculate a transition trajectory which starts at the old gait and ends with the endpoints of the new gait.
+
+        :param old_subgait:
+            The subgait which the transition is starting from
+        :param new_subgait:
+            The subgait which the transition will result in.
+
+        :returns:
+             list of joints, velocities and positions corresponding to every joint at given timestamps and duration
+        """
+        joints = []
+        transition_positions = []
+        transition_velocities = []
+        transition_durations = []
+
+        total_transition_factor = len(new_subgait[0].setpoints)
+        for transition_index in range(total_transition_factor):
+            rospy.logwarn('transition index {ti} with timestamp {ts}'
+                          .format(ti=str(transition_index), ts=str(new_subgait[0][transition_index].time)))
+
+            positions = []
+            velocities = []
+            for old_joint in old_subgait.joints:
+                new_joint = new_subgait.get_joint(old_joint.name)
+
+                if len(old_joint) != len(new_joint):
+                    rospy.logerr('Amount of setpoints is inconsistent between subgait {og} of gait {fg} and {sg}'
+                                 .format(og=old_subgait.subgait_name, fg=old_subgait.gait_name,
+                                         sg=new_subgait.gait_name))
+
+                joints.append(new_joint.name)
+
+                new_setpoint_factor = 1.0 / (total_transition_factor - 1) * transition_index
```

Comment:
```suggestion
                new_setpoint_factor = transition_index / (total_transition_factor - 1.0)
```
Avoids ambiguity in order of computation